tests/yam/migration/migration_unattended.pm

```diff
@@ -0,0 +1,58 @@
+# SUSE's openQA tests
+#
+# Copyright 2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Migration activation then reboot to perform migration.
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use grub_utils 'grub_test';
+use migration 'disable_installation_repos';
+use power_action_utils 'power_action';
+use utils 'zypper_call';
+
+sub await_migration {
+    my $timeout = 600;
+    my @tags = qw(migration-exit grub2);
+    while (1) {
+        my $ret = check_screen \@tags, 30;
+	$timeout -= 30;
+        diag("left total await_migration timeout: $timeout");
+	if (match_has_tag('grub2')) {
+            last;
+        }
+	if (match_has_tag('migration-exit')) {
+            wait_still_screen 30;
+            next;
+        }
+    }
+
+}
+sub run {
+    my $self = shift;
+
+    enter_cmd "echo 'url: " . get_var('SCC_URL') . "' > /etc/SUSEConnect";
+
+    select_console('root-console');
+
+    my $repo_home = "http://download.suse.de/ibs/home:/fcrozat:/SLES16/SLE_\$releasever";
+    my $repo_images = 'http://download.suse.de/ibs/home:/fcrozat:/SLES16/images/';
+    zypper_call("ar -p 90 '$repo_home' home_sles16");
+    zypper_call("ar -p 90 $repo_images home_images");
+
+    # install the migration image and active it
+    zypper_call("--gpg-auto-import-keys -n in suse-migration-sle16-activation");
+
+    power_action('reboot', keepconsole => 1, first_reboot => 1);
+
+    assert_screen('grub-menu-migration');
+    wait_screen_change { await_migration };
```

Comment:
so if in 10 seconds the screen doesn't change, then this would fail, the migration could be stuck for a while ...this might now work in some circumstances.

Reply:
updated